### PR TITLE
Remove Grammar Ambiguities

### PR DESCRIPTION
**Pull request by jdconrad**
Title plus a bit of clean up.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/LStatic.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.elasticsearch.painless.Definition;
+import org.elasticsearch.painless.Definition.Type;
+import org.elasticsearch.painless.Variables;
+import org.elasticsearch.painless.Variables.Variable;
+import org.objectweb.asm.Opcodes;
+import org.elasticsearch.painless.MethodWriter;
+
+/**
+ * Represents a variable load/store.
```
Comment (rmuir) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/node/LStatic.java]: is this really what it is?

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/BasicAPITests.java [hunk 2]
```diff
@@ -35,8 +33,6 @@ public void testSetIterator() {
             "int total = 0; while (y.hasNext()) total += y.next(); return total;"));
         assertEquals("abc", exec("Set x = new HashSet(); x.add(\"a\"); x.add(\"b\"); x.add(\"c\"); " +
             "Iterator y = x.iterator(); String total = \"\"; while (y.hasNext()) total += y.next(); return total;"));
-        assertEquals(3, exec("def x = new HashSet(); x.add(2); x.add(3); x.add(-2); def y = x.iterator(); " +
-            "def total = 0; while (y.hasNext()) total += (int)y.next(); return total;"));
     }
```
Comment (rmuir) [inline modules/lang-painless/src/test/java/org/elasticsearch/painless/BasicAPITests.java]: were these supposed to be removed?

**Comment by rmuir** [comment]
I love that this is fixed, thanks for doing all this work to do it the clean way, it will be worth it! I left minor comments but looks great. With antlr 4 this is the best we can do, fail on this stuff in tests, and test everything. See this explanation: https://groups.google.com/d/msg/antlr-discussion/z8otgVzqB7I/IAHEU9keZLgJ

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/LStatic.java (hunk 1 above)
Comment (jdconrad) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/node/LStatic.java]: Good catch.   Copy and paste error.  Fixed!

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/BasicAPITests.java (hunk 2 above)
Comment (jdconrad) [inline modules/lang-painless/src/test/java/org/elasticsearch/painless/BasicAPITests.java]: Oops!  Not sure why I did that.

**Comment by jdconrad** [comment]
@rmuir Thanks for the review and all the help debugging!  Glad you caught the issue.

**Comment by jdconrad** [comment]
This was merged, but squashed due to 30 unimportant commit messages.

**Pull request closed without merging**